### PR TITLE
removed `paths fetch` dependency for interchain/chains dir

### PR DESCRIPTION
**Pull request by boojamya**
`paths fetch` was reliant on chain info being added to the interchain/chains folder. This info is now being maintained by the chain registry. No need to have this as a dependency.

### CODE REVIEW

**Comment by boojamya** [comment]
In fact, is there a use for _interchain/chains_ folder anymore?
Shall I remove?

**Comment by jtieri** [comment]
> In fact, is there a use for _interchain/chains_ folder anymore? Shall I remove?

Assuming that we want to treat the `chain-registry` as the de facto source for chain metadata then yeah we can scrap this directory entirely.

In the code we were checking that a chain existed in `interchain/chains` before attempting to pull the path metadata, we just need to tweak this now to check that the path json file exists instead and make sure the cloned GitHub repo is being cleaned up after the cmd runs.

**Review by jtieri — APPROVED** [review]

File: cmd/paths.go [hunk 1]
```diff
@@ -284,29 +283,29 @@ $ %s pth fch`, appName, defaultHome, appName)),
 			// Try to fetch path info for each configured chain that has canonical chain/path info in the GH repo
 			for _, srcChain := range config.Chains {
 				for _, dstChain := range config.Chains {
-					fName := fmt.Sprintf("%s.json", srcChain.ChainID())
 
 					// Check that the constructed URL is valid
-					u, err := url.Parse(fmt.Sprintf("%s%s", jsonURL, fName))
+					// ex. https://github.com/cosmos/relayer/tree/main/interchain/{chain-id}
+					u, err := url.Parse(fmt.Sprintf("%s/%s", PATHSURL, srcChain.Chainid))
```
Comment (boojamya) [inline cmd/paths.go]: I think you are missing parenthesis here: srcChain.ChainId()))

**Comment by boojamya** [comment]
My thought process for removing lines 289 through 301 was that we already clone the repo down. We could just check to see if the path info exists when we try and read from the folder on line 307.

File: cmd/paths.go (hunk 1 above)
Comment (jtieri) [reply cmd/paths.go]: Good catch, I think technically this field is still being initialized on the chain struct but I think we will want to revise the chain struct and have everything behind the `ChainProvider` so best to use the method

**Comment by jtieri** [comment]
> My thought process for removing lines 289 through 301 was that we already clone the repo down. We could just check to see if the path info exists when we try and read from the folder on line 307.

You are 100% correct :man_facepalming: 

This will effectively do the same thing without having to make the http request
https://github.com/cosmos/relayer/blob/dan/path_fetch_patch/cmd/paths.go#L306-L310

**Pull request merged**